packages/preact-urql/src/hooks/useQuery.ts

```diff
@@ -90,7 +90,8 @@ function toSuspenseSource<T>(source: Source<T>): Source<T> {
 const isSuspense = (client: Client, context?: Partial<OperationContext>) =>
   client.suspense && (!context || context.suspense !== false);
 
-const sources = new Map<number, Source<OperationResult>>();
+type Sources = Map<number, Source<OperationResult>>;
+const sourcesByClient = new WeakMap<Client, Sources>();
```

Comment:
This will possibly heap a lot though so it might be worth to moving this to the `client` itself?

Reply:
Thanks for starting work on this bug so fast!!

My feeling is that putting these caches on the client instance would be the best bet. I saw this line and figure it could be related but I still don't fully understand the context in which this cache is used. Why wouldn't the cache or dedup exchanges take care of this case?

Reply:
I suppose this implementation detail may even change as we experiment with more implementations of suspense?

Regarding this PRs relation to #1271, why is this relevant if we don't wish to use this temporary cache on the server-side? Has something gone wrong in the SSR setup where it's accidentally active again?

Reply:
That is a good point, the error we're seeing is only because of the tests running in a browser environment. This wouldn't happen in a real node environment. 😅

Reply:
The tests I shared are running in a node sandbox.

Reply:
https://codesandbox.io/s/urql-cross-request-cache-issue-jbfws

Reply:
This also was not happening on previous versions. Only when I updated to the latest in a project I got similar tests to fail, so I created a standalone version of that and eventually ported that to codesandbox to be able to quickly share it. If you download the files from the sandbox and you run them in your computer (jest running in node) then you will see the tests failing too. 

I believe the problem is that this cache `const sources = new Map<number, Source<OperationResult>>();` is in the file scope, so once you imported the file it will be cached for as long as the node process lives. So in the case where you have a server running SSR that gets different requests you are likely to see the issues I represented in the test case.

Reply:
When you run your tests we see `typeof window === 'object'` which means it's being run in a browser context. The sources are unused in a Node context (`typeof window === 'undefined'`)

Reply:
Ohh so does it mean that because I mocked window.fetch the way I did then I am seeing the issues?

Reply:
Ohh, so it is probably because I mocked window.fetch the way I did that caused it to fail then? Interesting. I will update that on my tests and try it out.

Reply:
So I mean, following the principle of least surprise this probably wasn't a great choice of implementations on our part, but it's rather temporary since we'll also need to figure out other suspense edge cases

Reply:
Thanks for responding so quickly on this "bug". Makes me think that moving that cache to the client is the safest too, but I defer that to you :)